.github/workflows/caching.yml

```
@@ -182,25 +184,35 @@ jobs:
 
       - if: steps.compiled-deps.outputs.cache-hit != 'true'
         run: |
-          cabal update
+          cabal v2-update
+
+      - if: steps.compiled-deps.outputs.cache-hit != 'true' && runner.os == 'Linux' && matrix.ghc == '8.10.7'
+        name: Download sources for bench
+        #  Downloaded separately, to match the tested work/PR workflow guarantees
+        run: |
+          cabal $cabalBuild --enable-benchmarks --only-download
 
       - if: steps.compiled-deps.outputs.cache-hit != 'true'
-        name: Download all sources
+        name: Download the rest of the sources
+        #  Downloaded separately, to match the tested work/PR workflow guarantees
         run: |
-          cabal $cabalBuild --only-download
+          cabal $cabalBuild --enable-tests      --only-download
 
       # repeating builds to workaround segfaults in windows and ghc-8.8.4
       # This build agenda in not to have successful code,
       # but to cache what can be cached, so step is fault tolerant & would always succseed.
       # 2021-12-11: NOTE: Building all targets, since
       # current Cabal does not allow `all --enable-tests --enable-benchmarks --only-dependencies`
-      - if: steps.compiled-deps.outputs.cache-hit != 'true'
-        name: Build all targets; try 3 times
-        continue-on-error: true
+
+      - if: steps.compiled-deps.outputs.cache-hit != 'true' && runner.os == 'Linux' && matrix.ghc == '8.10.7'
+        name: (For Bench workflow) Build benchmark targets
```

Should not have this
```suggestion
        name: (For Bench workflow) Build benchmark targets
        continue-on-error: true
```
?